tests/run/fused_types.pyx

```diff
@@ -361,6 +362,22 @@ def test_fused_memslice_dtype_repeated_2(cython.floating[:] array1, cython.float
     """
     print cython.typeof(array1), cython.typeof(array2), cython.typeof(array3)
 
+def test_fused_const_memslice_dtype_repeated(const cython.floating[:] array1, cython.floating[:] array2):
```

Comment:
We probably also need more tests for arguments with plain (non-memoryview) fused types. For example, I'd be interested in how this behaves if a fused type is defined with both a const and non-const variant of a basic C type (e.g. `int` and `const int` in the same fused type). Not sure if this should be allowed, but there should better be a test for it.

Reply:
Since `const const int` leads to a "Duplicate 'const'" error during parsing, to me it would make sense that using `const` with a fused type containing a `const` type leads to an error.

Reply:
Agreed.
What I meant, though, was, that you could define a fused type with both `int` and `const int` in it, and then use it as such (without additional `const`). Admittedly, that isn't really related to this PR, but it's worth a test.

Reply:
This test:
```
cdef fused mix_const_t:
    int
    const int

cdef cdef_func_with_mix_const_type(mix_const_t val):
    print(val, cython.typeof(val))

def test_fused_with_mixed_const():
    """Test fused type including const and non-const types

    >>> test_fused_with_mixed_const()
    (0, 'int')
    """
    cdef int val = 0
    cdef_func_with_mix_const_type(val)
```
leads to the following errors at the line calling `cdef_func_with_mix_const_type`:
```
Invalid use of fused types, type cannot be specialized
ambiguous overloaded method
```

This looks to be because both implementations match the situation (https://github.com/cython/cython/blob/master/Cython/Compiler/PyrexTypes.py#L4444).

To me, there is not much sense to support it since `int` can be passed as `const int`, but that would probably be best to raise the error at the line of the fused type definition.

I'm adding it as an error test as it is. Let me know if this is OK.

Reply:
Seems ok for now, and good to have a separate error test for it, in case we want to report the error (read: a better error) earlier at some point.

Reply:
Still:
> We probably also need more tests for arguments with plain (non-memoryview) fused types.

Could you add just a few for that case, please?